sys/include/net/gcoap.h

```diff
@@ -446,6 +446,16 @@ extern "C" {
  */
 #define GCOAP_PAYLOAD_MARKER    (0xFF)
 
+/**
+ * @ingroup net_gcoap_conf
+ * @brief   Flag to delay initialization until explicit command
+ *
+ * By default gcoap starts at auto-initialization time.
+ */
+#ifndef GCOAP_INIT_DELAY
+#define GCOAP_INIT_DELAY        0
```

Comment:
I am a bit worried by the name. The first thing that came to mind when I looked at the PRs title without any description, was that this PR adds a *timed* delay to the gCoAP initialization somehow. Maybe call it `GCOAP_NO_AUTO_INIT`? Then again, we have something like this—just in reverse—with `gnrc_rpl`, where there is a dedicated `gnrc_rpl_auto_init` module, that ensures auto-initialization when complied in.

Reply:
I understand your confusion. I'm OK with `GCOAP_NO_AUTO_INIT`. @pokgak, works for you?

Reply:
yup OK